share/util/__init__.py

```diff
@@ -124,7 +124,7 @@ def __visit(self, node):
 
         self.__visiting.add(key)
         for k in self.__dependencies(node):
-            if k is not None:
+            if k is not None and k is not key:
```

Comment:
What in what case is does `k == key`? Having a dependency on yourself should be considered an error.